src/drivers/queue_interop/Queue.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace yii\queue\queue_interop;
+
+use Interop\Amqp\AmqpContext;
+use Interop\Queue\PsrConnectionFactory;
+use Interop\Queue\PsrContext;
+use Interop\Queue\PsrProducer;
+use yii\base\NotSupportedException;
+use yii\queue\cli\Queue as CliQueue;
+
+class Queue extends CliQueue
+{
+    /**
+     * {@inheritdoc}
+     */
+    public $commandClass = Command::class;
+
+    /**
+     * @var string
+     */
+    public $factoryClass = null;
+
+    /**
+     * @var array
+     */
+    public $factoryConfig = [];
+
+    /**
+     * @var string
+     */
+    public $queueName = 'queue';
+
+    /**
+     * @var PsrContext
+     */
+    private $context;
+
+    /**
+     * @var PsrProducer
+     */
+    private $producer;
+
+    /**
+     * Listens queue and runs new jobs.
+     */
+    public function listen()
+    {
+        $consumer = $this->getContext()->createConsumer(
+            $this->getContext()->createQueue($this->queueName)
+        );
+
+        while (true) {
```

Comment:
I was wrong. There is no need for sleep since `receive` is blocking (it waits on a socket, or does sleep internally). No need to do it here.